pkg/controllers/deploymentreplicassyncer/deployment_replicas_syncer_controller.go

```diff
@@ -0,0 +1,165 @@
+
+	deployment := &appsv1.Deployment{}
+	binding := &workv1alpha2.ResourceBinding{}
+	bindingName := names.GenerateBindingName(util.DeploymentKind, req.Name)
+
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: req.Namespace, Name: bindingName}, binding); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("no need to update deployment replicas for deployment not found")
+			return controllerruntime.Result{}, nil
+		}
+		return controllerruntime.Result{}, err
+	}
+
+	// if it is not divided schedule type, no need to update replicas
+	if binding.Spec.Placement.ReplicaSchedulingType() != policyv1alpha1.ReplicaSchedulingTypeDivided {
+		return controllerruntime.Result{}, nil
+	}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, deployment); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("no need to update deployment replicas for binding not found")
+			return controllerruntime.Result{}, nil
+		}
+		return controllerruntime.Result{}, err
+	}
+
+	// if replicas in spec already the same as in status, no need to update replicas
+	if deployment.Spec.Replicas != nil && *deployment.Spec.Replicas == deployment.Status.Replicas {
+		klog.Infof("replicas in spec field (%d) already equal to in status field (%d)", *deployment.Spec.Replicas, deployment.Status.Replicas)
+		return controllerruntime.Result{}, nil
+	}
+
+	// make sure the replicas change in deployment.spec can sync to binding.spec, otherwise retry
+	if deployment.Spec.Replicas == nil || *deployment.Spec.Replicas != binding.Spec.Replicas {
+		klog.Errorf("wait until replicas of binding (%d) equal to replicas of deployment (%d)",
+			binding.Spec.Replicas, *deployment.Spec.Replicas)
+		return controllerruntime.Result{}, fmt.Errorf("retry to wait replicas change sync to binding")
+	}
+
+	// make sure the scheduler observed generation equal to generation in binding, otherwise retry
+	if binding.Generation != binding.Status.SchedulerObservedGeneration {
+		klog.Errorf("wait until scheduler observed generation (%d) equal to generation in binding (%d)",
+			binding.Status.SchedulerObservedGeneration, binding.Generation)
+		return controllerruntime.Result{}, fmt.Errorf("retry to wait scheduler observed generation")
+	}
+
+	if len(binding.Status.AggregatedStatus) != len(binding.Spec.Clusters) {
+		klog.Errorf("wait until all clusters status collected, got: %d, expected: %d",
+			len(binding.Status.AggregatedStatus), len(binding.Spec.Clusters))
+		return controllerruntime.Result{}, fmt.Errorf("retry to wait status in binding collected")
+	}
+	for _, status := range binding.Status.AggregatedStatus {
+		if status.Status == nil {
+			klog.Errorf("wait until aggregated status of cluster %s collected", status.ClusterName)
+			return controllerruntime.Result{}, fmt.Errorf("retry to wait status in binding collected")
+		}
+	}
+	klog.Infof("all %d clusters status collected success", len(binding.Status.AggregatedStatus))
```

Comment:
The information described in this log may be incorrect. It is valid only when some new clusters are propagated.